File: 0/Templates/VisualBasic/ProjectTemplates/SplitViewAppVBTemplate/Presentation/NotifyPropertyChanged.vb
Namespace Presentation
    Public Class NotifyPropertyChanged
        Implements INotifyPropertyChanged

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Protected Sub OnPropertyChanged(<CallerMemberName> Optional propertyName As String = Nothing)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub

        Public Function SetProperty(Of T)(ByRef storage As T, value As T, <CallerMemberName> Optional propertyName As String = Nothing) As Boolean
            If Not Object.Equals(storage, value) Then
                storage = value
                OnPropertyChanged(propertyName)
                Return True
            End If
            Return False
        End Function
    End Class
End Namespace

File: 0/Templates/CSharp/ProjectTemplates/ContentAppTemplate/Presentation/NotifyPropertyChanged.cs
using System;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace $safeprojectname$.Presentation
{
    public abstract class NotifyPropertyChanged : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;

        protected void OnPropertyChanged([CallerMemberName]string propertyName = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

        public bool Set<T>(ref T storage, T value, [CallerMemberName()]string propertyName = null)
        {
            if (!object.Equals(storage, value)) {
                storage = value;
                OnPropertyChanged(propertyName);
                return true;
            }
            return false;
        }
    }
}
